src/site/content/en/blog/google-search-sw/index.md

```diff
@@ -0,0 +1,463 @@
+but as of this writing, the approach is
+[experimental](https://developers.google.com/web/updates/2018/09/asynchronous-access-to-http-cookies#origin-trial)
+and is not widely supported.)
+
+A mismatch between the service worker's view of the current logged in user and
+the actual user logged in to the Google Search web interface could lead to
+incorrectly personalized search results or misattributed metrics and logging.
+Any of those failure scenarios would be a serious issue for the Google Search
+team.
+
+**Solution**
+
+Rather than wait for experimental APIs to launch and provide direct access to
+the browser's cookies inside of a service worker, the Google Search team went
+with a stop-gap solution: whenever a page controlled by the service worker is
+loaded, the page will read the relevant cookies and use
+[`postMessage()`](https://developer.mozilla.org/en-US/docs/Web/API/Worker/postMessage)
+to send them to the service worker.
+
+The service worker can then check the current cookie value against the value
+that it expects, and if there's a mismatch, it will take steps to purge any
+user-specific data from its storage, and reload the search results page without
+any incorrect personalization.
+
+The specific steps that the service worker takes to reset things to a baseline
+are particular to Google Search's requirements, but the same general approach
+may be useful to other developers who deal with personalized data keyed off of
+browsers cookies.
+
+### Experiments and dynamism
+
+As mentioned, the Google Search team relies heavily on running experiments in
+production, and testing the effects of new code and features in the real world
+before turning them on by default. This can be a bit of a challenge with a
+static service worker that relies heavily on cached data, since opting users in
+and out of experiments often requires communication with the backend server.
+
+**Solution**
+
+The solution that the team went with was to use a dynamically generated service
+worker script, customized by the web server for each individual user, instead of
+a single, static service worker script that gets generated ahead of time.
+Information about experiments that might affect the service worker's behavior or
+network requests in general can be included directly in this customized service
+worker script, and changing the sets of experiments that are active for a user
+can be done via a combination of traditional techniques, like browser cookies,
+as well as serving updated code in the registered service worker URL.
+
+Using a dynamically generated service worker script also makes it easier to
+provide an escape hatch in the unlikely event that a service worker
+implementation has a fatal bug that needs to be avoided. The folks responsible
+for the Google Search service worker can have the dynamic service worker
```

Comment:
Suggestion to change the focus from the team to the actual change, so instead of 'The folks...', try:

The dynamic server worker response includes a ...